### PR TITLE
doc/user: New Sidebar Structure

### PR DESCRIPTION
**Pull request by ruf-io**
### Motivation

This is a draft update of the new documentation sidebar structure. Many tasks are incomplete, do not merge!

Preview Link: https://preview.materialize.com/11566/

### Functionality Changes

- [ ] **Sidebar menu scrolls independent of main page** - This makes it easier to scan and navigate from page-to-page without losing your spot in the current doc.
- [ ] **Current page is highlighted and visible** - When you navigate to a new page, it's highlighted in the sidebar, and the sidebar is scrolled so that the selected item is in the vertical center of the page
- [ ] **Top-level indexes are _mostly_ clickable** - Clicking the top-level category page (like overivew) takes you to a page.
- [ ] **Search moves to top of page** - Everything else about it works the same.

### Structural Changes

Where ever possible, structural changes were superficial-only in order to change as few URLs as possible

- [ ] **s/demos/quickstarts/** - We're moving away from using `demo` and towards `quickstart` for two reasons: (1) quickstarts have a more interactive "you're getting hands-on with it" connotation, and (2) Users should be able to actually take (in its entirety or in parts) the GitHub repo behind each quickstart and use it as scaffolding for their Materialize use case.
- [ ] **Move SQL, cloud, CLI, Idioms (previously part of guides) to a new Reference section in menu** - the urls don't change for the most part, but this helps highlight important reference pages (like JOINS) that were previously lost in the sea of SQL.
- [ ] **Move most guides into Integrations, in Client Libraries and Integration Guides** - We always intended to add some structure to guides when we had enough content. 

### Follow-on Tasks

Several placeholder pages are created and marked with draft: true in front matter, these will need to be published as a follow-up

### What to review

Functionality - Any glitchy behavior with the menu? Any browser compatibility issues? Responsiveness issues?

Content Structure - Anything specifically wrong, misleading, or worse than before?

### CODE REVIEW

**Comment by morsapaes** [comment]
This looks so much better, thanks for working on it, @ruf-io! Some comments as I walked through it:

* **Overview:** `Speedrun: Materialize a view` is way less straightforward than `Get started` (especially for non-native speakers). I'd keep that, and maybe switch up the order to be `Key concepts`->`Get started`->`Advanced concepts`. I'm missing having something like `Connect` as more high-level section; at least to me this gets a bit lost under `Reference`.

* **Quickstarts:** The naming isn't consistent, so we should either use `Build sth` or `Materialize for sth`; but not a mix. Do we aim to keep this section to the core use cases we want to push? It feels a little heavy right now, so we might want to keep it to ~3 good demos and put use cases that are still a little green (_e.g._ Materialize as a feature store) under `More use cases`? Also, I think the SQL Spellbook would fit better in this section!

* **Reference:** It's cool that you unearthed the most relevant sections from under that pile of SQL commands! This makes a lot of sense. I'd excise all the `SQL` subsections into it's own...section? Or just have that as a single sub-section called `SQL`, and have the breakdown within a landing page there (since it's not really relevant to have stuff like `SQL Data Types`, `SQL Identifiers` or `SQL Namespaces` as top-level sections). Eventually, we should figure out a sane way to organize all the SQL commands into groups to get rid of this insane list.

There's a `connections`->`Sources` menu seemingly lost at the bottom, too! Let me know what you think about the points above and I can directly push some suggestions!

**Review by morsapaes — CHANGES_REQUESTED** [review]
(See above)

File: doc/user/config.toml [hunk 1]
```diff
@@ -67,91 +67,89 @@ versions = [
...
+weight= 70
 
 [[menu.main]]
-name = "SQL Overview"
-parent = "sql"
-url = "/sql/"
-weight = 1
+name = "Subscriptions (`TAIL`)"
+parent = "reference"
+url = "/sql/create-index/"
```
Comment (morsapaes) [inline doc/user/config.toml:102]:
```suggestion
url = "/sql/tail/"
```

File: doc/user/content/data-input-formats.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: "Data Input Formats"
```
Comment (morsapaes) [inline doc/user/content/data-input-formats.md:2]: Is this section needed? This information will live in the refurbished landing page for sources that I'm pushing out this week(-ish)!

File: doc/user/content/metrics.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
+title: "Metrics"
```
Comment (morsapaes) [inline doc/user/content/metrics.md:2]: Should go under "Operating Materialize"?

Comment (morsapaes) [reply doc/user/content/metrics.md:2]: Same for `System Catalog`.

**Comment by ruf-io** [comment]
>Overview: Speedrun: Materialize a view is way less straightforward than Get started (especially for non-native speakers). I'd keep that, and maybe switch up the order to be Key concepts->Get started->Advanced concepts. 

Good point, Done!

**Comment by ruf-io** [comment]
>I'm missing having something like Connect as more high-level section; at least to me this gets a bit lost under Reference.

Let's do a fast-follow PR and figure out the best place to put that information.

**Comment by ruf-io** [comment]
> Quickstarts: The naming isn't consistent, so we should either use Build sth or Materialize for sth; but not a mix.

Edited `SQL on Streaming Logs` to "Run SQL on Streaming Logs" so that each quickstart starts with a verb

**Comment by ruf-io** [comment]
>  [Quickstarts] feels a little heavy right now, so we might want to keep it to ~3 good demos and put use cases that are still a little green (e.g. Materialize as a feature store) under More use cases? 

You're probably looking at it with drafts rendered. (`hugo serve -D`) here's what it'll look like in prod (`hugo serve`)
image

It's still a little awkward right now because I am using the existing demos, but I think as soon as we overhaul the demos to align with the use cases we can put the most important ones at the top and deemphasize the others.

File: doc/user/content/integrations/psql.md [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
+aliases:
+  - /connect/
+  - /connect/cli/
 menu:
   main:
-    parent: "connections"
-    weight: 1
+    parent: "integrations"
+    weight: 5
+    name: "`psql` CLI"
```
Comment (joacoc) [inline doc/user/content/integrations/psql.md:11]: `psql` looks small in the navbar in Safari but looks good in Chrome.
The same with "`materialized` Binary"

Screen Shot 2022-04-13 at 14 48 34

**Comment by joacoc** [comment]
Maybe not necessary now, but the cloud is hidden last in the reference. It is essential for the platform but may not be so much from a documentation perspective inside the reference section. It's like a tradeoff between importance and utility. 

It shares topics with other sections like `Operating Materialize` (Troubleshooting, Monitoring) 
Screen Shot 2022-04-13 at 14 57 26

**Comment by ruf-io** [comment]
> Maybe not necessary now, but the cloud is hidden last in the reference. It is essential for the platform but may not be so much from a documentation perspective inside the reference section. It's like a tradeoff between importance and utility.
> 
> It shares topics with other sections like `Operating Materialize` (Troubleshooting, Monitoring) Screen Shot 2022-04-13 at 14 57 26

When platform hits GA, we can remove most of this section and just integrate it into the rest of docs.

File: doc/user/content/integrations/psql.md (hunk 4 above)
Comment (ruf-io) [reply doc/user/content/integrations/psql.md:11]: good catch!

File: doc/user/content/metrics.md (hunk 3 above)
Comment (ruf-io) [reply doc/user/content/metrics.md]: deleted metrics.

Comment (ruf-io) [reply doc/user/content/metrics.md]: I see the division as:

image

Operating Materialize is all "How-To" and System catalog is more reference

File: doc/user/content/integrations/psql.md (hunk 4 above)
Comment (ruf-io) [reply doc/user/content/integrations/psql.md:11]: fixed by switching to fira code font 
image

**Review by joacoc — APPROVED** [review]

**Pull request merged**